app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreenViewModel.kt

```diff
@@ -357,9 +357,9 @@ class OtherUserProfileScreenViewModel @Inject constructor(
     }
 
     override fun onBlockUser(blockUserState: BlockUserDialogState) {
-        viewModelScope.launch(dispatchers.io()) {
+        viewModelScope.launch {
             requestInProgress = true
-            when (val result = blockUser(userId)) {
+            when (val result = withContext(dispatchers.io()) { blockUser(userId) }) {
```

Comment:
Not related to this line, but we should do the same for `updateConversationMutedStatus` in the function above.
Switching dispatchers to call the use case :D